core/src/main/scala/org/apache/spark/sql/execution/joins/LocalJoin.scala

```diff
@@ -290,9 +362,11 @@ case class LocalJoin(leftKeys: Seq[Expression],
     val relationTerm = ctx.freshName("relation")
     val relationIsUnique = ctx.freshName("keyIsUnique")
     val buildRDDRef = ctx.addReferenceObj("buildRDD", buildRDD)
-    val buildPartRef = ctx.addReferenceObj("buildPartition", buildPartition)
-    ctx.addMutableState(classOf[HashedRelation].getName, relationTerm,
-      prepareHashedRelation(ctx, relationTerm, buildRDDRef, buildPartRef))
+    if (buildRDD.partitions.size == 1) {
```

Comment:
Use replicatedTableJoin flag. Though doesn't matter since the non-optimized versions are due tol be removed soon.